src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -945,10 +999,66 @@ public override MethodInfo[] GetMethods(BindingFlags bindingAttr)
         }
 
         [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.NonPublicFields)]
-        public override FieldInfo? GetField(string name, BindingFlags bindingAttr) => throw new NotSupportedException();
+        public override FieldInfo? GetField(string name, BindingFlags bindingFlags)
+        {
+            ArgumentNullException.ThrowIfNull(name);
+            ThrowIfNotCreated();
+
+            FieldInfo? match = null;
+            StringComparison compare = (bindingFlags & BindingFlags.IgnoreCase) != 0 ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
+            foreach (FieldBuilderImpl fieldInfo in _fieldDefinitions)
+            {
+                BindingFlags fieldFlags = GetBindingFlags(fieldInfo);
+                if (name.Equals(fieldInfo.Name, compare) && (bindingFlags & fieldFlags) == fieldFlags)
+                {
+                    if (match != null)
+                    {
+                        if (ReferenceEquals(fieldInfo.DeclaringType, match.DeclaringType))
+                        {
+                            throw new AmbiguousMatchException(SR.Format(SR.AmbiguousMatch_MemberInfo, fieldInfo.DeclaringType, fieldInfo.Name));
+                        }
+                    }
+
+                    match = fieldInfo;
```

Comment:
I can't recall the logic here, but I would normally assume we should break after finding something. However, Reflection does odd things so I could also see it looping over all of them "just in case". If a `break` isn't appropriate I would comment that we are intentionally looping over the entire collection. Bonus points for the why.

Reply:
Followed the existing runtime implementation which loops through all and throws when duplicates found, I will add comment.